### PR TITLE
Homework assistant

### PR DESCRIPTION
**Pull request by Dantemss**
# Done
- Homework assistant basics
- Assistant Specs
- API
- API Specs
- API to return a list of assignable exercises
- Add description to homework task

### CODE REVIEW

**Comment by Dantemss** [comment]
All done.

**Comment by pandafulmanda** [comment]
Adding ```rake``` instructions from @Dantemss for convenience =)

```rake db:drop db:create db:migrate db:seed sprint:009homework```

**Comment by Dantemss** [comment]
I realized a couple of days ago that we need an API to return all assignable exercises...

**Comment by jivewise** [comment]
I believe I saw comps that indicated that the homeworks have a description field.  @philschatz?

**Comment by Dantemss** [comment]
That's easy to add if needed.

**Comment by philschatz** [comment]
UX mockups have it. Is it necessary @fredasaurus?

**Comment by Fredasaurus** [comment]
yes. *probably

**Comment by Dantemss** [comment]
Still needs homework description. Also, merge #148 first.

**Comment by Dantemss** [comment]
OK really done now...

**Comment by Dantemss** [comment]
OK REALLY should be done now :)

**Comment by Dantemss** [comment]
Description is part of the "settings" for the plan btw.

File: app/representers/api/v1/task_representer.rb [hunk 1]
```diff
@@ -6,33 +6,48 @@ class TaskRepresenter < Roar::Decorator
...
 
     property :due_at,
              type: DateTime,
-             writeable: true,
+             writeable: false,
              readable: true,
              schema_info: {
-               required: true,
-               description: "When the task is due (nil means not due)"
+               description: "When the task is due (nil means always open)"
```
Comment (jpslav) [inline app/representers/api/v1/task_representer.rb]: Really nil means never closes (there could be an `opens_at` which would mean it wasn't "always" open)

**Comment by jpslav** [comment]
@Dantemss looks great - thanks! some minor comments but I'm going to go ahead and merge.

**Comment by Dantemss** [comment]
I'll push the description fix to master... Can't break anything changing a string, right? :D

**Comment by Dantemss** [comment]
Or I'll just make a small PR.

**Pull request merged**